common/connect/src/main/java/com/csgroup/s2pdgs/repository/FileRepository.java

```
@@ -20,6 +20,15 @@ public record FileRepository(Path basePath) implements Repository<FileRepository
         return Arrays.stream(files).map(FileEntry::new);
     }
 
+    @Override
+    public Stream<Entry> walk(String path) throws IOException {
+        final var root = Path.of(basePath.toString(), normalizePath(path));
+        return Files.walk(root)
+            .filter(it -> !it.equals(root))
+            .map(Path::toFile)
+            .map(FileEntry::new);
+    }
+
```

Much more readable than the default one but hide it's behavior. Force to connect a webdav or ftp server to detect bug on default function because file repository doesn't share anymore the same code. Few chance to have a bug on this function so you can keep it.